### PR TITLE
Add black check to Python files in CI

### LINKED ISSUE #230: Add `black` check to `.py` files in lint CI
Currently, we are not checking the formatting of the Python files on CI, meaning that we won't be made aware of any formatting issues if we don't check that locally.

Add `black -l 120` to the Python files on the repo so this can be checked automatically.

### PR DESCRIPTION
**Pull request by anipaul2**
Fixes #230 

**Description**
This PR adds a new job in the CI that checks the formatting of Python files using Black. This ensures that all Python files follow the same code styling guidelines, making the codebase cleaner and easier to read.

**Changes**
1. Modified `.github/workflows/build.yaml` to include a new job for Black checks.

Please let me know if any changes are required.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,5 +3,6 @@
 members = [
     "teos",
     "teos-common",
-    "watchtower-plugin"
+    "watchtower-plugin",
+    "misbehaviour_tool/backend"
```
Comment (mariocynicys) [inline Cargo.toml:7]: This shouldn't be here.

**Review by mariocynicys — CHANGES_REQUESTED** [review]
That comment, looks good otherwise.

**Comment by mariocynicys** [comment]
Also this should be a one commit PR, and [sign the commit](https://help.github.com/en/github/authenticating-to-github/signing-commits).

**Comment by anipaul2** [comment]
> Also this should be a one commit PR, and [sign the commit](https://help.github.com/en/github/authenticating-to-github/signing-commits).

I thought of signing when all the checks get successfully passed.

**Comment by mariocynicys** [comment]
Rebase on the latest master and give a short commit message like ``Use black for `.py` files formatting``, the commit message you are using right now would be nice as a description instead.

**Comment by anipaul2** [comment]
> Rebase on the latest master and give a short commit message like `` Use black for `.py` files formatting ``, the commit message you are using right now would be nice as a description instead.

There is a option which shows update branch and two list which are:-

1. Update with merge commit
2. Update with rebase

Are you suggesting me to select the second option?

**Comment by mariocynicys** [comment]
> There is a option which shows update branch and two list which are

Neither of these. I mean rebasing `master` on your branch locally and then force pushing on this feature branch.
For this you want to:
- Make sure you have the remote `master` up to date: `git fetch --all`.
- Rebase on the remote's master: `git rebase REMOTE_NAME/master` (for instance, I have my REMOTE_NAME as `talaia`, which points to `https://github.com/talaia-labs/rust-teos`, you want to pick the remote name you gave for that).
- Fix any conflicts, I guess in this case there won't be any.
- Sign the commit you introduced by amending it.
- Force push, and you will be connected to `master`.

**Comment by anipaul2** [comment]
> > There is a option which shows update branch and two list which are
> 
> Neither of these. I mean rebasing `master` on your branch locally and then force pushing on this feature branch. For this you want to:
> 
> * Make sure you have the remote `master` up to date: `git fetch --all`.
> * Rebase on the remote's master: `git rebase REMOTE_NAME/master` (for instance, I have my REMOTE_NAME as `talaia`, which points to `https://github.com/talaia-labs/rust-teos`, you want to pick the remote name you gave for that).
> * Fix any conflicts, I guess in this case there won't be any.
> * Sign the commit you introduced by amending it.
> * Force push, and you will be connected to `master`.

Done. Thanks for the point-to-point explanation.

**Review by mariocynicys — APPROVED** [review]
LGTM

File: .gitignore
Comment (sr-gi) [inline .gitignore:1]: Updating the .gitignore won't be needed for this.

**Review by sr-gi — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -46,4 +46,12 @@ jobs:
...
\ No newline at end of file
+          cargo clippy --all-features --all-targets --color always -- --deny warnings
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+      - name: Install black
+        run: pip install black
+      - name: Check Python formatting
+        run: black . --check -l 120
```
[the remark this answers is not in the document]
Comment (anipaul2) [reply .github/workflows/build.yaml:57]: > We can do this in a simpler way directly using the black GitHub Actions from https://black.readthedocs.io/en/stable/integrations/github_actions.html
> 
> I'm wondering also wether we should add this to the same job, or create a new one called `python-lint`.

Creating a new one would be better. Should I do like that?

File: .gitignore
Comment (anipaul2) [reply .gitignore:1]: okay.

File: .github/workflows/build.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sr-gi) [reply .github/workflows/build.yaml:57]: I think it'd be cleaner if so. What's your opinion @mariocynicys?

[the remark this answers is not in the document]
Comment (mariocynicys) [reply .github/workflows/build.yaml:57]: Aha, I'm pro separating this into a different job as well.

[the remark this answers is not in the document]
Comment (anipaul2) [reply .github/workflows/build.yaml]: Done.

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -46,4 +46,17 @@ jobs:
...
+          cargo clippy --all-features --all-targets --color always -- --deny warnings
+  python-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
```
Comment (mariocynicys) [inline .github/workflows/build.yaml:58]: This won't be needed.

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -46,4 +46,17 @@ jobs:
...
+      - name: Install black
```
Comment (mariocynicys) [inline .github/workflows/build.yaml:59]: Call this `Run black` instead.

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -46,4 +46,17 @@ jobs:
...
+        uses: psf/black@stable
+        with:
+          args: "watchtower-plugin/tests --check -l 120"
```
Comment (mariocynicys) [inline .github/workflows/build.yaml:62]: These are called `options` not `args`.
Also there is a dedicated parameter called `src` which I guess selects what sources you wanna format.
as per `https://black.readthedocs.io/en/stable/integrations/github_actions.html`

**Review by mariocynicys — COMMENTED** [review]
Some hints

File: .gitignore
Comment (mariocynicys) [reply .gitignore:1]: I find a lot of mac users add this thing in their .gitignore s all the time, I think having it there won't hurt and would avoid future contributors pushing that file by mistake.

File: .github/workflows/build.yaml (hunk 3 above)
Comment (anipaul2) [reply .github/workflows/build.yaml:58]: okay.

File: .github/workflows/build.yaml (hunk 4 above)
Comment (anipaul2) [reply .github/workflows/build.yaml:59]: okay.

File: .github/workflows/build.yaml (hunk 5 above)
Comment (anipaul2) [reply .github/workflows/build.yaml:62]: I see, using dedicated parameters is the better approach. I'll change args to options and specify the src in the next commit.

File: .github/workflows/build.yaml [hunk 6]
```diff
@@ -46,4 +46,14 @@ jobs:
           cargo fmt --verbose --check -- --color always
       - name: Run clippy
         run: |
-          cargo clippy --all-features --all-targets --color always -- --deny warnings
\ No newline at end of file
+          cargo clippy --all-features --all-targets --color always -- --deny warnings
+  python-lint:
```
Comment (sr-gi) [inline .github/workflows/build.yaml:50]: Add a line break here

**Review by sr-gi — CHANGES_REQUESTED** [review]

**Review by sr-gi — APPROVED** [review]

**Comment by sr-gi** [comment]
Needs rebase

**Pull request merged**